### DISCUSSION TITLE
How is task.id deterministic?

### DISCUSSION THREAD

**Discussion by gterras** [vitest-dev/vitest#5746] [Q&A] [answered]
Hi I would like to known where or how exactly is `task.id` is generated, since I rely on it for my snapshot file names. Thanks for your answers.

**Comment by sheremet-va** [comment, marked as answer]
It's based on the test filepath, project name and the position inside the suite:

`aptsb64bs` would be a file path plus project name hash

`aptsb64bs_0` is the first suite

`aptsb64bs_0_0` is the first test inside the first suite

**Reply by gterras** [reply]
Thank you that's clear, a few thoughts : 
- I have test ids starting with `-` I don't know if it's expected
- Having the `0_0_etc` part also available in the task object as a formatted array (like `[0, 1, 0]`) could be useful for `toMatchFileSnapshot()` usage (since most of the time snapshots are saved in the current test folder the unique id part isn't really useful)

**Reply by sheremet-va** [reply]
Having `-` at the start is fine, it's part of the file+projectName hash
